docs/design/verification_protocol.md

```diff
@@ -0,0 +1,117 @@
+# Public Health Authority Diagnosis Verification Protocol
+
+This design covers the exposure notifications server's ability to verify
+diagnosis certifications from public health authorities.
+
+The actual process of issuing these certificates is not covered in this
+document, but will be at a later date.
+
+## Motivation
+
+The current exposure notifications publication protocol only has the ability
+to authenticate that a request came from a mobile device via the vendor
+specific device attestation protocols. This doesn't give us the ability
+to certify that the diagnosis came from a trusted (by the server) public
+health authority in the jurisdiction.
+
+## Definitions
+
+* PHA : Public Health Authority
+* JWT : JSON Web Token
+* ECDSA : Elliptic Curve Digital Signature Algorithm
+* PIN : personal identification number. Could be a different authenticator.
+* HMAC : hash-based message authentication code
+* TEK : Temporary Exposure Key
+
+
+## High Level Flow
+
+1. App user is diagnosed with Covid-19. PHA issues a PIN code to the user.
+2. The user enters the PIN code on their App
+3. The app creates a HMAC value that combines the TEKs from the device along
+   with calculated transmission risk value, rolling period start and rolling
+	 period count values (the key data + metadata).
+	 These values are sent to the PHA server that issued the PIN.
+4. If the PIN is valid, the PHA issues a JWT that is signed using their
+   ECDSA p-256 algorithm. The JWT includes additional claims about the data
+	 (see below).
+5. The app on the user's device sends this signed JWT to the exposure
+   notifications server (this project).
+6. If the JWT is valid and signed by a trusted PHA (verified because the public
+	 key has been previously shared with the server), then the keys are imported
+	 into the server for distribution to other devices in the geography.
+
```

Comment:
Should we store any of this verification alongside the data?

Reply:
That's a good question.

We could store the `iss` field, as in which authority issued this diagnosis. 

We could also build in storage of the `phadata` - but as the reference sever won't make use of it, I lean against storing it. 

Good debate to be had.

Reply:
I don't like the idea of the TEK server being used to "lookup" a verification, but it could be useful in the opposite direction (given a TEK, from where was it verified, when, etc)

Reply:
I'm inclined to agree that we shouldn't sore this in the reference